src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs

```diff
@@ -97,7 +97,11 @@ public override double ViewportValue
         /// <inheritdoc/>
         public override Size MeasureOverride(Size availableSize)
         {
-            var window = Owner.GetVisualRoot() as TopLevel;
+            var maxAvailableSize = (Owner.GetVisualRoot() as WindowBase)?.PlatformImpl?.MaxClientSize;
+            if (!maxAvailableSize.HasValue)
+            {
+                maxAvailableSize = (Owner.GetVisualRoot() as TopLevel)?.ClientSize;
```

Comment:
I've only added the logic that will be executed if there is *no* window available. In this case it tries to obtain information from another type of toplevel.